Runtimes/Core/cmake/modules/EmitSwiftInterface.cmake

```
@@ -12,6 +12,16 @@ function(emit_swift_interface target)
       $<$<COMPILE_LANGUAGE:Swift>:-emit-private-module-interface-path$<SEMICOLON>${CMAKE_CURRENT_BINARY_DIR}/$<TARGET_PROPERTY:${target},Swift_MODULE_NAME>.private.swiftinterface>
       $<$<COMPILE_LANGUAGE:Swift>:-library-level$<SEMICOLON>api>
       $<$<COMPILE_LANGUAGE:Swift>:-Xfrontend$<SEMICOLON>-require-explicit-availability=ignore>)
+
+      # Emit catalyst swiftmodules and interfaces
+      if(SwiftCore_VARIANT_MODULE_TRIPLE)
+        set(variant_module_tmp_dir "${CMAKE_CURRENT_BINARY_DIR}/${target}-${SwiftCore_VARIANT_MODULE_TRIPLE}")
```

Why in the new code we are not using the module name property like for the "regular" module files, e.g.

```suggestion
        set(variant_module_tmp_dir "${CMAKE_CURRENT_BINARY_DIR}/$<TARGET_PROPERTY:${target},Swift_MODULE_NAME>-${SwiftCore_VARIANT_MODULE_TRIPLE}")
```

---

`file(MAKE_DIRECTORY ...)` can't take generator expressions, so that doesn't work.
`add_custom_command(OUTPUT ...)` doesn't allow target-dependent generator expressions, so `$<TARGET_PROPERRY:...>` doesn't work.